### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Ruslan-Mykhalchak**

### CODE REVIEW

**Review by Saharij — APPROVED** [review]
Great!!!🔥

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
```
Comment (aliceblunt8) [inline src/herbivoresAndCarnivores.js:25]: fix according to the task

`Herbivore has a method of hide, which changes the hidden property of the beast to the opposite value and helps to hide from carnivores.`

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
```
Comment (aliceblunt8) [inline src/herbivoresAndCarnivores.js:30]:
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
```

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,38 @@
...
+      animal.health -= 50;
+
+      Animal.animalAlive();
```
Comment (aliceblunt8) [inline src/herbivoresAndCarnivores.js:33]: filter array on condition `if the health of the animal reaches 0`

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**